heartbeat/monitors/active/tcp/tcp_test.go

```diff
@@ -60,14 +76,47 @@ func TestUpEndpoint(t *testing.T) {
 				"tcp",
 				"up",
 			),
-			hbtest.TCPChecks(port),
+			hbtest.RespondingTCPChecks(port),
 			mapval.Schema(mapval.Map{
 				"resolve": mapval.Map{
 					"host":   "localhost",
 					"ip":     "127.0.0.1",
 					"rtt.us": mapval.IsDuration,
 				},
 			}),
-		))(event.Fields),
+		)),
+		event.Fields,
+	)
+}
+
+func TestConnectionRefusedEndpointJob(t *testing.T) {
+	ip := "127.0.0.1"
+	port := uint16(freeport.GetPort())
+	event := testTCPCheck(t, ip, port)
+
+	mapvaltest.Test(
+		t,
+		mapval.Strict(mapval.Compose(
+			tcpMonitorChecks(ip, ip, port, "down"),
+			hbtest.ErrorChecks(fmt.Sprintf("dial tcp %s:%d: connect: connection refused", ip, port), "io"),
+			hbtest.TCPBaseChecks(port),
+		)),
+		event.Fields,
+	)
+}
+
+func TestUnreachableEndpointJob(t *testing.T) {
+	ip := "203.0.113.1"
+	port := uint16(1234)
+	event := testTCPCheck(t, ip, port)
+
+	mapvaltest.Test(
+		t,
+		mapval.Strict(mapval.Compose(
+			tcpMonitorChecks(ip, ip, port, "down"),
+			hbtest.ErrorChecks(fmt.Sprintf("dial tcp %s:%d: i/o timeout", ip, port), "io"),
```

Comment:
have you considered to introduce `ErrorChecksf`?

```
func ErrorCheckfs(str strings, vs ...interface{}) {
```

Reply:
That would clearly be better in this test, but I'd like to avoid that at the moment. Currently the signature is:

`func ErrorChecks(msg interface{}, errType string) mapval.Validator `

`msg` is not `string`, but rather `interface()` so that a user can specify either a string value or a `mapval.IsDef`. While in this specific case `ErrorChecksfs` would be better, I think it's better to have fewer methods that are more generic here. I don't think it'd be used enough to justify the extra surface area.